scipy/optimize/tests/test_chandrupatla.py

```diff
@@ -82,7 +83,6 @@ def callback(res):
     return _chandrupatla_wrapper
 
 
-_chandrupatla_root = _wrap_chandrupatla(find_root)
```

Comment:
When `find_root` was added, we didn't take the time to update the tests written for the `_chandrupatla` interface to the new interface. Instead, this translates calls to "`_chandrupatla_root'` to calls to `find_root` (which ultimately calls `_chandrupatla`). This just ensured that everything worked via the new interface.

Actually replacing the calls to `_chandrupatla_root` with direct calls to `find_root` was one of the things I wanted to do before we begin to add other `method`s, so I've gone ahead and done that here. Perhaps that part should just be a separate PR?

Reply:
It looks like that was all changed in one commit so I'd be happy to review a pr that cherry picks just that commit

Reply:
Sure I'll do that.

Reply:
gh-22088